src/chat-handler.js

```diff
@@ -0,0 +1,28 @@
+
+const chatHandler = (route, data) => {
+  switch (route) {
+    case ('CHATROOM.REGISTER') : {
+      newuser = new NewUser(data);
```

Comment:
This switch case will be called when a new socket is registering itself (against an existing endpoint in our data store). So a student journey could look like this;
1. Student registers themselves when first visiting the site (post request to /join)
2. Server registers the user/endpoint in rooms module
3. Server sends client view
4. Client side js fires Websocket for the first time (msg with chatroom.register)
5. Server adds the sockets id to the rooms object